arch/arc/core/dsp/dsp_offsets.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2022 Synopsys.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ARCv2 DSP and AGU structure member offset definition file
+ *
+ */
+#ifdef CONFIG_ARC_DSP_SHARING
+GEN_OFFSET_SYM(_callee_saved_stack_t, dsp_ctrl);
+GEN_OFFSET_SYM(_callee_saved_stack_t, acc0_lo);
```

Comment:
The acc0_lo and hi registers are aliases for r58 and r59, which are already saved/restored in Zephyr when CONFIG_ARC_HAS_ACCL_REGS is defined, or CONFIG_64BIT. I think we need some more clever CONFIG and macro definitions to avoid saving these registers multiple times.
With splitting off the DSP stuff in a different file, it is even harder to spot such duplication.

Reply:
Probably it's enough to select CONFIG_ARC_HAS_ACCL_REGS if CONFIG_ARC_DSP_SHARING is set.